src/wp-includes/link-template.php

```diff
@@ -1811,9 +1807,30 @@ function get_adjacent_post( $in_same_term = false, $excluded_terms = '', $previo
 
 	$current_post_date = $post->post_date;
 
-	$join     = '';
-	$where    = '';
 	$adjacent = $previous ? 'previous' : 'next';
+	$order    = $previous ? 'DESC' : 'ASC';
+
+	$args = array(
+		'posts_per_page'         => 1,
+		'author__in'             => array(),
```

Comment:
I'm not sure this is needed as the default for `author__in` is already empty.

Reply:
Because I do this later on. 

```
$args['author__in'][]  = $user_id;
```